roles/os_net_setup/tasks/main.yml

```diff
@@ -1,53 +1,90 @@
-      ansible.builtin.fail:
-        msg: |
-          Failed to Retrive auth Infomation
-          clouds.yaml: {{ clouds_yaml }}
-          osp_secret: {{ osp_secret }}
+    - name: Delete existing networks
+      ansible.builtin.shell: |
+        set -euxo pipefail
+        if [
+            $(oc exec -n openstack openstackclient --
+              openstack network list -c Name -f value |
+              grep -c {{ item.name }}) != 0
+        ];then
+          oc exec -n openstack openstackclient --
+          openstack network delete {{ item.name }}
+        fi
+      loop: "{{ cifmw_os_net_setup_config }}"
+      when: not cifmw_os_net_setup_dry_run | bool
+    - name: Print network creation commands
+      ansible.builtin.debug:
+        msg: "{{ lookup('ansible.builtin.template', _template_file) }}"
+    - name: Create networks
+      ansible.builtin.shell: >
+        {{ lookup('ansible.builtin.template', _template_file) }}
+      when: not cifmw_os_net_setup_dry_run | bool
 
-- name: Get network information
-  openstack.cloud.networks_info:
-    auth: "{{ openstack_auth }}"
-    validate_certs: "{{ cifmw_os_net_setup_verify_tls }}"
-    ca_cert: /etc/pki/tls/certs/ca-bundle.crt
-  register: cifmw_os_net_setup_network_list_out
-  until: cifmw_os_net_setup_network_list_out is not failed
-  retries: "{{ cifmw_os_net_setup_osp_calls_retries }}"
-  delay: "{{ cifmw_os_net_setup_osp_calls_delay }}"
+- name: Create subnet resources
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  vars:
+    _template_file: 'subnet_command.j2'
+  block:
+    - name: Delete existing subnets
+      ansible.builtin.shell: |
+        set -euxo pipefail
+        if [
+            $(oc exec -n openstack openstackclient --
+              openstack subnet list -c Name -f value |
+              grep -c {{ item.1.name }}) != 0
+        ];then
+          oc exec -n openstack openstackclient --
+          openstack subnet delete {{ item.1.name }}
+        fi
+      loop: >-
+        {{
+          cifmw_os_net_setup_config |
+          subelements('subnets', skip_missing=True)
+        }}
+      when: not cifmw_os_net_setup_dry_run | bool
+    - name: Print subnet command creation
+      ansible.builtin.debug:
+        msg: "{{ lookup('ansible.builtin.template', _template_file) }}"
+    - name: Create subnets
+      ansible.builtin.shell: >
+        {{ lookup('ansible.builtin.template', _template_file) }}
+      when: not cifmw_os_net_setup_dry_run | bool
 
-- name: Process network list element
-  ansible.builtin.include_tasks:
-    file: subtask_net.yml
-  loop: "{{ cifmw_os_net_setup_config|flatten(levels=1) }}"
-  loop_control:
-    loop_var: net_item
+- name: Create required subnet pools
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  vars:
+    _template_file: 'subnet_pool_command.j2'
+  block:
+    - name: Delete existing subnet pools
```

Comment:
I'm not sure if that will work as expected. If you will have existing resources created already, and You will first delete old subnets and create new ones, it may use existing subnet pool which will cause failure in the deletion of the subnet pool in this step.
To make it working correctly I think You should have 2 blocks:
1. to delete all existing resources,
2. to create new resources as requested

Reply:
makes sense. Done :)